libraries/generic/source/generate.js

```diff
@@ -16,20 +16,30 @@ function proxy (value, generator, segments = []) {
   if (typeof value !== 'object' || value === null || Array.isArray(value)) return value
 
   return new Proxy(value, {
-    get: (_, key) => {
-      segments.push(key)
+    get: (node, key) => {
+      if (key in node) return original(node, key)
 
-      const value = generator(segments)
+      const next = [...segments, key]
+      const value = generator(next)
 
-      return proxy(value, generator, segments)
+      return proxy(value, generator, next)
     },
     set: (_, key, value) => {
-      segments.push(key)
-      generator(segments, value)
+      const next = [...segments, key]
+
+      generator(next, value)
 
       return true
     }
   })
 }
 
+function original (node, key) {
+  const value = node[key]
+
+  // breaks fn.length
+  if (typeof value === 'function') return (...args) => value.apply(node, args)
```

Comment:
why not fn.bind ?

Reply:
Because I'm too old for `function.bind` :)